src/lib/support/StringBuilder.h

```diff
@@ -89,13 +91,27 @@ class StringBuilderBase
     }
 };
 
-/// a preallocated sized string builder
-template <size_t kSize>
+/// A preallocated sized string builder
+/// Default buffer size is 257
+template <size_t kSize = 257>
 class StringBuilder : public StringBuilderBase
 {
 public:
+    /// Default constructor
     StringBuilder() : StringBuilderBase(mBuffer, kSize) {}
 
+    /// Constructor for char * and length
+    StringBuilder(const char * data, size_t length) : StringBuilder() { AddFormat("%.*s", static_cast<int>(length), data); }
+
+    /// Constructor for uint8_t * and length
+    StringBuilder(const uint8_t * data, size_t length) : StringBuilder(Uint8::to_const_char(data), length) {}
```

Comment:
This casting to char* should be handled by consumers, so they are clear that they are treating bytes (which may not be printable) as chars.  We should not hide that conversion.